parser/util.go

```diff
@@ -19,6 +22,30 @@ func ResolvePath(root, path string) (string, error) {
 	return filepath.Join(root, path), nil
 }
 
+// ExpandHomeDir expands the path to include the home directory if the path is
+// prefixed with `~`. If it isn't prefixed with `~`, the path is returned as-is.
+func ExpandHomeDir(path string) (string, error) {
+	if len(path) == 0 {
+		return path, nil
+	}
+
+	if path[0] != '~' {
+		return path, nil
+	}
+
+	if len(path) > 1 && path[1] != '/' && path[1] != '\\' {
+		return "", errors.New("cannot expand user-specific home dir")
+	}
+
+	// Non-cgo required, available since go1.12
```

Comment:
Non-cgo required means this only works with non-cgo builds? Is that right?

Reply:
Yes, sorry the language is confusing. It works without cgo.